packages/react-juce/src/components/Canvas.ts

```diff
@@ -1,66 +1,164 @@
+  }
+
+  fillRoundedRect(x: number, y: number, width: number, height: number, cornerSize: number): void {
+    this._drawCommands.push(['fillRoundedRect', x, y, width, height, cornerSize]);
+  }
+
+  clearRect(x: number, y: number, width: number, height: number): void {
+    this._drawCommands.push(['clearRect', x, y, width, height]);
+  }
+
+  //================================================================================
+  // Path functions
+  //
+  // TODO: Should we split things out into CanvasRenderingContextPath
+  //       so things like close()/stroke() are more obvious when looking at API.
+  //       If you look at the ts type definitions for CanvasRenderingContext2D
+  //       CanvasRenderingContext2D is composed of other objects like CanvasRenderingContextPath
+  //       which contains all path methods. What is the best way to do this in JS and share
+  //       a drawCommands instance?
+  beginPath(): void {
+    this._drawCommands.push(['beginPath']);
+  }
+
+  lineTo(x: number, y: number): void {
+    this._drawCommands.push(['lineTo', x, y]);
+  }
+
+  moveTo(x: number, y: number): void {
+    this._drawCommands.push(['moveTo', x, y]);
+  }
+
+  arc(x: number, y: number, radius: number, startAngle: number, endAngle: number): void {
+    //TODO: Add support for optional antiClockWise?: boolean arg
+    this._drawCommands.push(['arc', x, y, radius, startAngle, endAngle]);
+  }
+
+  quadraticCurveTo(cpx: number, cpy: number, x: number, y: number): void {
+    this._drawCommands.push(['quadraticCurveTo', cpx, cpy, x, y]);
+  }
+
+  closePath(): void {
+    this._drawCommands.push(['close']);
+  }
+
+  stroke(): void {
+    this._drawCommands.push(['stroke']);
+  }
+
+  fill(): void {
+    this._drawCommands.push(['fill']);
+  }
+
+  //================================================================================
+  // Transform functions
+  rotate(angle: number): void {
+    this._drawCommands.push(['rotate', angle]);
+  }
+
+  translate(x: number, y: number): void {
+    this._drawCommands.push(['translate', x, y]);
+  }
+
+  setTransform(a: number, b: number, c: number, d: number, e: number, f: number): void {
+    this._drawCommands.push(['setTransform', a, b, c, d, e, f]);
+  }
+
+  resetTransform(): void {
+    this._drawCommands.push(['resetTransform']);
+  }
+
+  //================================================================================
+  // Image functions
+  //
+  //TODO: Add support for other drawImage overloads.
+  //      Currently only support SVG string. What is correct
+  //      type to use here?
+  drawImage(image: string, dx: number, dy: number): void {
+      this._drawCommands.push(['drawImage', image, dx, dy]);
+  }
+
+  //================================================================================
+  // Text functions
+  strokeText(text: string, x: number, y: number, maxWidth?: number): void {
+    if (maxWidth === undefined)
```

Comment:
Might be better here to go with
```
if (typeof maxWidth === 'number')
  // pass it along
else
  // dont pass it
```

Seems more appropriately explicit as a condition for when we want to pass it along.

I don't want to hold up the merge on account of this though, so if you're down for this change let's come back for it in the follow up PR